cpp/src/slice2java/Gen.cpp

```diff
@@ -4780,8 +4780,18 @@ Slice::Gen::ServantVisitor::visitInterfaceDefEnd(const InterfaceDefPtr& p)
     if (!allOps.empty())
     {
         out << sp;
-        out << nl << "@Override" << nl
-            << "default java.util.concurrent.CompletionStage<com.zeroc.Ice.OutgoingResponse> dispatch("
+        out << nl << "/**";
+        out << nl
+            << " * Dispatches an incoming request to one of the methods of this generated interface, based on the "
+               "operation name carried by the request.";
+        out << nl << " *";
+        out << nl << " * @param request the incoming request";
+        out << nl << " * @return the outgoing response";
+        out << nl
+            << " * @throws com.zeroc.Ice.UserException when the response payload contains a {@code UserException}.";
```

Comment:
I think it should be more like, when a user exception is thrown Ice marshals it as the response payload. The The current sentence sounds like the response payload already contains a exception and we throw the exception in this case.